forwarder/rpc_decompressor.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 The Vulcan Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package forwarder
+
+import (
+	"io"
+	"io/ioutil"
+
+	"github.com/golang/snappy"
+)
+
+// Decompressor represents object that implements a GRPC Compressor interface.
+// Currently only supports Snappy decompression.
+type Decompressor struct{}
```

Comment:
Add a `var _ grpc.Decompressor = &Decompresser{}` to assert that you implement the interface at compile-time.